### PR TITLE
Add/relative units for font size

### LINKED ISSUE #23323: Add relative units for font size option
The custom font size option is really useful, but the lack of relative units has been a problem to make patterns work well in both large and small screens.

It'd be great if we could have some the relative units, just like the cover block height control.

Screenshot 2020-06-19 at 18 39 24 

With the `vw` unit, this sort of type treatment would be possible, I think.

Screenshot 2020-06-19 at 18 41 34

I've seen this topic has come up before, and I'll apologise if there is a better thread to discuss this. If there is please point me at there.

### PR DESCRIPTION
**Pull request by kirilzh**
## Description
Fixes: #23323 
This PR adds the option to set custom CSS font sizes to font-size-picker component by adding a switch toggle underneath `Font sizes` dropdown. 

Currently, when the user sees a dropdown with the predefined font-sizes, an adjacent `custom` field which accepts only numbers and shows the font size using pixels for units of measure, and the reset button which clears the custom field and sets font size to `default`. 

To allow for writing more complex sizing formulas in this PR adds a toggle which when activated removes the font-size dropdown and renders a large custom field which in addition accepts string values.

## How has this been tested?
This is a draft still so will add tests when changes are accepted.

## Screenshots 
Toggle OFF.
This is the default state of the toggle. 

image  

Toggle ON.

image

## Types of changes
New feature (non-breaking change which adds functionality).

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by joyously** [comment]
I have a difficult time determining which is OFF and which is ON, especially since the one you say is OFF has a box that says Custom.

**Comment by kirilzh** [comment]
@joyously I updated the description. You're right that in both cases it the label is `Custom` but I kept it only because the text placeholder suggests that the user can write more expressive formulas. I would be happy to change it if you have something more appropriate in mind.

**Comment by joyously** [comment]
Can you show how it looks now? My main concern is that a rounded box with a circle in it does not indicate ON or OFF to me. How about a checkbox, which is fairly standard and easy to understand? Or just a value for Custom in the dropdown, that then allows input of the custom value.

**Comment by kirilzh** [comment]
@joyously Here is how the sizing options currently look:

with-regular-sizing

and here's how it looks with the changes from this PR

with-advanced-sizing

I think the toggle is fairly standard way of representing the state of the component. For example the "airplane mode" toggle on mobile devices is now ubiquitous, so my assumptions is that the users will find it familiar and easy to use.

**Comment by joyously** [comment]
Thanks for the screenshots.
I think checkboxes are more standard than that rounded thingy. Even though you think it's ubiquitous, I don't have much exposure to it. I do have experience with trying to help an elderly man use his phone. He doesn't have a clue what to do. And some time back, I helped a friend configure his replacement phone. We couldn't figure out how to get the mail app to check for new mail since there was nothing to interact with. I asked my nephew who told me to slide down from the top, and that everyone knows that. It did work, but there was no way for the user to figure that out.

The thing you think is standard might be common, but it certainly isn't friendly. It looks to me like a strange button because of the color, but that's only when ON. I can't tell if the blue is indicating the current state or where to click to activate it. It's more cognitive load than is necessary.
I'm no a11y expert, but I thought that things appearing and disappearing was bad for a11y.

**Comment by AlexChariyska** [comment]
I agree with @kirilzh. I believe that using the toggle button will be more consistent with the current design of the settings box, the Drop cap functionality uses the same type of toggle button as seen in the provided screenshots.

**Comment by joyously** [comment]
Yes, it's the same as others, which are also bad. But this is the one asking for design feedback so I mentioned it here.

File: packages/components/src/font-size-picker/index.js [hunk 1]
```diff
@@ -118,6 +148,16 @@ export default function FontSizePicker( {
 					{ __( 'Reset' ) }
 				</Button>
 			</div>
+			<div>
+				<ToggleControl
+					label="Advanced font sizing"
```
Comment (aristath) [inline packages/components/src/font-size-picker/index.js:153]: Text should be translatable

```suggestion
					label={ __( 'Advanced font sizing' ) }
```

File: packages/components/src/font-size-picker/index.js [hunk 2]
```diff
@@ -118,6 +148,16 @@ export default function FontSizePicker( {
...
+					onChange={ () =>
+						setAdvancedFontSizing( ! advancedFontSizing )
+					}
+					help="Toggle to activate custom font-sizing settings"
```
Comment (aristath) [inline packages/components/src/font-size-picker/index.js:157]: Should be translatable

```suggestion
					help={ __( 'Toggle to activate custom font-sizing settings' ) }
```

File: packages/components/src/font-size-picker/index.js [hunk 3]
```diff
@@ -68,11 +79,40 @@ export default function FontSizePicker( {
...
+				id: fontSizePickerNumberId,
+				className: classnames(
+					'components-font-size-picker__number',
+					advancedFontSizing &&
+						'components-font-size-picker__number--full-size'
+				),
+				onChange: ( event ) => onChange( event.target.value ),
+				ariaLabel: __( 'Custom' ),
+				value: value || '',
+				placeholder: 'e.g. calc(2rem - 13.5px)',
```
Comment (aristath) [inline packages/components/src/font-size-picker/index.js:93]: Should be translatable

```suggestion
				placeholder: __( 'e.g. calc(2rem - 13.5px)' ),
```

**Review by aristath — CHANGES_REQUESTED** [review]

File: packages/components/src/font-size-picker/index.js (hunk 3 above)
Comment (kirilzh) [reply packages/components/src/font-size-picker/index.js]: updated in `ade76846140e112e01ba2d1c09e8e9053940ac0b`

**Comment by aristath** [comment]
@kirilzh could you please rebase this PR? I wanted to check it out again today and see how we can move it forward, but the font-size-picker has changed a bit since the PR was first submitted so it will have to be updated first 👍

**Comment by paaljoachim** [comment]
Hi @aristath I believe that @kirilzh is not around any longer which means someone else will need to take the reins of this PR.

**Comment by aristath** [comment]
Closing this one since it no longer applies.
We still need to do it, but with the introduction of G2 components for font-size, this PR is no longer relevant and we'll need to fix this in G2 instead.

**Pull request closed without merging**